src/TLuaInterpreter.cpp

```diff
@@ -2522,6 +2522,27 @@ int TLuaInterpreter::killTrigger(lua_State* L)
     return 1;
 }
 
+int TLuaInterpreter::remainingTime(lua_State* L)
+{
+    string luaSendText = "";
+    if (!lua_isstring(L, 1)) {
+        lua_pushstring(L, "remainingTime: remainingTime requires a string ID");
+        lua_error(L);
+        return 1;
+    } else {
+        luaSendText = lua_tostring(L, 1);
+    }
+    Host& host = getHostFromLua(L);
+    QString text(luaSendText.c_str());
```

Comment:
There is little point in capturing the timer name as a `std::string` and then converting it to a `QString` - also - given that permanent timers can (or should) have names that are not confined to the Latin1 character encoding, whilst the `QString` constructor does natively convert from a `std::string` using it's `QString::fromUtf(...)` method that only works if `QT_NO_CAST_FROM_ASCII` is not `#define`d - which we might well do in the future; IMHO you should really be capturing the text directly into the `QString`.

All in all, I would suggest revising this function to:
```
int TLuaInterpreter::remainingTime(lua_State* L)
{
    if (!lua_isstring(L, 1)) {
        lua_pushfstring(L, "remainingTime: bad argument #1 type (timer name as string expected, got %s!", luaL_typename(L, 1));
        return lua_error(L);
    }
    QString timerName = QString::fromUtf8(lua_tostring(L, 1));
    Host& host = getHostFromLua(L);
    int result = host.getTimerUnit()->remainingTime(timerName);
    if (host.remainingTime(text) == -1) {
        lua_pushnil(L);
        lua_pushstring(L, "timer is inactive or expired");
        return 2;
    }
    if (host.remainingTime(text) == -2) {
        lua_pushnil(L);
        lua_pushstring(L, "timer \"%s\" not found", timerName.toUtf8().constData());
        return 2;
    }
    lua_pushnumber(L, result / 1000.0);
    return 1;
}
```
to complete my recommendation you then:
+ remove the changes to `Host` class (i.e. remove the changes to `Host.cpp` and `Host.h`)
+ revise the `TimerUnit` code to return `-2` if the `TTimer` is not found (change the `return -1;` at the end of `(int) TimerUnit::remainingTime(const QString&)`

HTH